WordPress/Classes/ViewRelated/Blog/CreateNewBlog/CreateNewBlogViewController.m

```diff
@@ -0,0 +1,551 @@
+#import "CreateNewBlogViewController.h"
+#import "WordPressComApi.h"
+#import "WPNUXMainButton.h"
+#import "WPNUXSecondaryButton.h"
+#import "WPWalkthroughTextField.h"
+#import "WPAsyncBlockOperation.h"
+#import "WPComLanguages.h"
+#import "WPWalkthroughOverlayView.h"
+#import "WPNUXUtility.h"
+#import "WPStyleGuide.h"
+#import "UILabel+SuggestSize.h"
+#import "WPAccount.h"
+#import "Blog.h"
+#import "WordPressComServiceRemote.h"
+#import "AccountService.h"
+#import "BlogService.h"
+#import "ContextManager.h"
+#import "NSString+XMLExtensions.h"
+#import "Constants.h"
+
+#import "WordPress-Swift.h"
+
+@interface CreateNewBlogViewController ()<UITextFieldDelegate,UIGestureRecognizerDelegate> {
+    WPNUXSecondaryButton *_cancelButton;
+    UILabel *_titleLabel;
+    UILabel *_siteAddressWPComLabel;
+    WPWalkthroughTextField *_siteTitleField;
+    WPNUXMainButton *_createBlogButton;
+    WPWalkthroughTextField *_siteAddressField;
+    
+    NSOperationQueue *_operationQueue;
+    
+    BOOL _authenticating;
+    BOOL _keyboardVisible;
+    BOOL _userDefinedSiteAddress;
+    CGFloat _keyboardOffset;
+    
+    NSDictionary *_currentLanguage;
+}
+
+@end
+
+@implementation CreateNewBlogViewController
+
+static CGFloat const CreateBlogStandardOffset             = 15.0;
+static CGFloat const CreateBlogTextFieldWidth             = 320.0;
+static CGFloat const CreateBlogTextFieldHeight            = 44.0;
+static CGFloat const CreateBlogTextFieldPhoneHeight       = 38.0;
+static CGFloat const CreateBlogiOS7StatusBarOffset        = 20.0;
+static CGFloat const CreateBlogButtonWidth                = 290.0;
+static CGFloat const CreateBlogButtonHeight               = 41.0;
+
+static UIEdgeInsets const CreateBlogCancelButtonPadding     = {1.0, 0.0, 0.0, 0.0};
+static UIEdgeInsets const CreateBlogCancelButtonPaddingPad  = {1.0, 13.0, 0.0, 0.0};
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _operationQueue = [[NSOperationQueue alloc] init];
+        _currentLanguage = [WPComLanguages currentLanguage];
+    }
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.view.backgroundColor = [WPStyleGuide wordPressBlue];
+    
+    [self initializeView];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillShow:)
```

Comment:
@kwonye you'll need to do a cleanup call on the `dealloc` method of this class to remove the controller as an observer within the `NSNotificationCenter` via `[[NSNotificationCenter defaultCenter] removeObserver:self];`. Otherwise, the observation will likely crash once this view is deallocated.